src/shared/logger.ts

```diff
@@ -100,13 +102,13 @@ export function getLogger(): Logger {
  * you need to call releaseLogger. This will end the ability to write to the logfile with this logger instance.
  */
 export function createLogger(params: LoggerParams): Logger {
-    let level: string
+    let level: LogLevel
     if (params.logLevel) {
         level = params.logLevel
-    } else {
+    } else { // level is required in LoggerParams which is required, so else can never happen
```

Comment:
In that case, just remove the `else`.

Reply:
I'd prefer to have a chat with @bryceitoc9 in case he had other plans when he wrote that in case he had other intensions.

Reply:
@micholas Hmm, thought I made it optional. Did it this way (assuming this was optional) in case getting the `SettingsConfiguration` was expensive. Your call on how to move forward from here, either should work.

Reply:
@bryceitoc9 I'm fine with making level optional. Actually I like that option, since it lets logger initialize itself (be responsible for logging stuff) as discussed in issue [353](https://github.com/aws/aws-toolkit-vscode/issues/353).

Reply:
`LoggerParams.level` is not required. Comment removed.
✔️